tests/gradients/test_parameter_shift_cv.py

```diff
@@ -0,0 +1,1105 @@
+            qml.Kerr(1.0, wires=[1])
+            qml.expval(qml.P(0))
+            qml.expval(qml.X(1))
+
+        # First rotation gate has no succeeding non-Gaussian operation
+        assert _grad_method(tape, 0) == "A"
+        # Second rotation gate does no succeeding non-Gaussian operation
+        assert _grad_method(tape, 1) == "F"
+        # Kerr gate does not support the parameter-shift rule
+        assert _grad_method(tape, 2) == "F"
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.Rotation(1.0, wires=[1])
+            # entangle the modes
+            qml.Beamsplitter(1.0, 0.0, wires=[0, 1])
+            # Non-Gaussian
+            qml.Kerr(1.0, wires=[1])
+            qml.expval(qml.P(0))
+            qml.expval(qml.X(1))
+
+        # After entangling the modes, the Kerr gate now succeeds
+        # both initial rotations
+        assert _grad_method(tape, 0) == "F"
+        assert _grad_method(tape, 1) == "F"
+        assert _grad_method(tape, 2) == "F"
+
+    def test_probability(self):
+        """Probability is the expectation value of a
+        higher order observable, and thus only supports numerical
+        differentiation"""
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(0.543, wires=[0])
+            qml.Squeezing(0.543, 0, wires=[0])
+            qml.probs(wires=0)
+
+        assert _grad_method(tape, 0) == "F"
+        assert _grad_method(tape, 1) == "F"
+        assert _grad_method(tape, 2) == "F"
+
+    def test_variance(self):
+        """If the variance of the observable is first order, then
+        parameter-shift is supported. If the observable is second order,
+        however, only finite-differences is supported."""
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.var(qml.P(0))  # first order
+
+        assert _grad_method(tape, 0) == "A"
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.var(qml.NumberOperator(0))  # second order
+
+        assert _grad_method(tape, 0) == "F"
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.Rotation(1.0, wires=[1])
+            qml.Beamsplitter(0.5, 0.0, wires=[0, 1])
+            qml.var(qml.NumberOperator(0))  # second order
+            qml.expval(qml.NumberOperator(1))
+
+        assert _grad_method(tape, 0) == "F"
+        assert _grad_method(tape, 1) == "F"
+        assert _grad_method(tape, 2) == "F"
+        assert _grad_method(tape, 3) == "F"
+
+    def test_second_order_expectation(self):
+        """Test that the expectation of a second-order observable forces
+        the gradient method to use the second-order parameter-shift rule"""
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.expval(qml.NumberOperator(0))  # second order

```

Comment:
In line 160 above, `var(qml.NumberOperator(0))` is marked as second-order, but here `expval` is, should this be `var` instead?

Reply:
Oops - this one is correct since `<n> = <X^2 + P^2 - hbar>` is second order in X and P. However, `Var(n) = <n^2> - <n>^2` is thus 4th order and the comment on line 160 is incorrect